### ISSUE TITLE
Notebook Terminal UX

### ISSUE THREAD

**Issue by sourishkrout** [stateful/vscode-runme#333]
Needs further prioritization. As per @mxsdev comments: Discussed and this is what we think is needed before release:

 - [x] Persistent terminal state (content, maybe cursor position)
 - [x] Slideout functionality (want after #289 because this may have the same issue)
 - [x] Decide exactly when to show integrated terminal by default (only for background tasks, user configurable, on all interactive tasks)
 - [x] Put behind feature flag if we decide to merge/release pre-emptively
 - [x] Focusing on terminal should not require two clicks (I think listen for iframe onFocus and direct that to terminal focus)
 - [x] Race condition where first line of output can miss the terminal - we need to probably wait for a "ready" message before running the program
 
Nice-to-haves:

 - [x] Pseudo-terminal sizing - should we use size of VSCode terminal or size of integrated terminal? (There is a conflict here; I think we should use the integrated terminal when it is focused and use the VSCode terminal when it is active/has focus) This may be required for running some TUIs in the integrated terminal, so that it can determine the proper sizing
with foreground over light background "mixed colors")
 - [x] Avoid using `!important` in styling if possible
 - [x] Font to be consistent with VSCode's terminal (both size and spacing)
 - [x] Proper cursor color (can be incorrect on light themes as pointed out by sebastian)
 - [x] Additional customizability options to user, styling, font, colors, etc
 - [x] Scrollbar consistent with VSCode terminal
 - [x] Short click sometimes doesn't focus window
 - [x] Resizing is broken
 - [x] Copy button (from old terminal output)

_Originally posted by @mxsdev in https://github.com/stateful/vscode-runme/issues/287#issuecomment-1468418218_

**Comment by sourishkrout** [comment]
We should break off a task for the portion of the notebook terminal that we want to expose without turning on a feature flag. 

As discussed, a "read-only" notebook terminal that mirrors the output of background tasks. This will help further demystify "background tasks" while they are running and reduce the UX-interaction surface to a minimum, so we take the appropriate time to work out read/write terminals and their relationship to the vscode-native terminal panel.

@degrammer @mxsdev: does that make sense?

**Comment by mxsdev** [comment]
@sourishkrout Makes sense to me - made #334

I'll assign it to @degrammer since I think we discussed that

**Comment by degrammer** [comment]
Yes, it makes sense! thank you for putting this together 🙏🏽
